File: scripting-style/templates/linear-analysis.R
# 当前问题：TODO_QUESTION
# 从项目根启动 scripts/analysis.R；已在脚本目录时改为 setwd(".")。
# 入口只运行一次，后续片段复用当前会话。
setwd("scripts")
getwd()
.libPaths()

library(Matrix)

## 读取表达矩阵 ====
# 本例使用矩阵 RDS；查看类型、标识和数值，再决定如何处理。
data_path <- "../data"
fn <- file.path(data_path, "TODO_COUNTS.rds")
counts <- readRDS(fn)
class(counts)
dim(counts)
counts[seq_len(min(5, nrow(counts))), seq_len(min(5, ncol(counts))), drop = FALSE]

## 读取样本信息 ====
# 先看实际字段和记录，为后续样本对齐提供依据。
fn <- file.path(data_path, "TODO_METADATA.tsv")
meta <- read.delim(fn, check.names = FALSE)
class(meta)
head(meta)
names(meta)

## 根据观察继续分析 ====
# 在此写入当前结果支持的下一步，尚未明确的分组和方法留待判断。
# 需要为下游保留已检查的 meta 时，可在相应片段使用：
# out_path <- "../output"
# dir.create(out_path, showWarnings = FALSE, recursive = TRUE)
# fn <- file.path(out_path, "metadata.checked.rds")
# saveRDS(meta, fn)

File: scripting-style/templates/linear-analysis.py
"""TODO: 当前科学问题，以及上游输入来自哪里。"""

# %% 工作目录
# 从项目根启动 scripts/analysis.py；若已在文件目录，将目标改为 "."。只运行一次。
import os
from pathlib import Path

os.chdir("scripts")
print(Path.cwd())

import pandas as pd

# %% 读取样本信息
# 查看字段、类型和少量记录，再确定后续分组。
data_path = Path("../data")
fn = data_path / "TODO_METADATA.tsv"
metadata = pd.read_csv(fn, sep="\t")

print(type(metadata))
print(metadata.head())
print(metadata.columns)

# %% 根据观察继续
# TODO：检查实际输出后续写下一步，选择依据放在调用附近。

# 后续确需这份表时，可在对应片段保存；格式与列名沿用项目约定。
# out_path = Path("../results/TODO_STEP")
# out_path.mkdir(parents=True, exist_ok=True)
# fn = out_path / "metadata.tsv"
# metadata.to_csv(fn, sep="\t", index=False)
